File: java/domain/src/main/java/com/youknow/domain/usecase/impl/GetMovieUsecase.java
package com.youknow.domain.usecase.impl;

import com.youknow.domain.model.Movie;
import com.youknow.domain.repository.MoviesRepository;
import com.youknow.domain.usecase.GetMovie;

public class GetMovieUsecase implements GetMovie {

    private MoviesRepository mMoviesRepository;

    public GetMovieUsecase(MoviesRepository moviesRepository) {
        mMoviesRepository = moviesRepository;
    }

    @Override
    public void get(String movieId, final GetMovieCallback callback) {
        mMoviesRepository.getMovie(movieId, new MoviesRepository.MovieLoadedCallback() {
            @Override
            public void onMovieLoaded(Movie movie) {
                callback.onMovieLoaded(movie);
            }
        });
    }
}

File: kotlin-coroutine-aac/app/src/main/java/com/youknow/movie/domain/usecase/impl/GetMovieUsecase.kt
package com.youknow.movie.domain.usecase.impl

import com.youknow.movie.domain.model.Movie
import com.youknow.movie.domain.repository.MoviesRepository
import com.youknow.movie.domain.usecase.GetMovie

class GetMovieUsecase(private val moviesRepository: MoviesRepository) : GetMovie {

    override suspend fun get(id: String): Movie = moviesRepository.getMovie(id)

}